Plugins/Test/Test.cpp

```diff
@@ -0,0 +1,291 @@
+#include "Test.hpp"
+
+#include "Utils/String.hpp"
+#include "External/rang/rang.hpp"
+#include "Services/Plugins/Plugins.hpp"
+#include "Services/Config/Config.hpp"
+
+#include "API/API/CExoFile.hpp"
+#include "API/API/CExoStringList.hpp"
+#include "API/API/CScriptCompiler.hpp"
+#include "API/API/CExoResMan.hpp"
+#include "API/API/CNWSModule.hpp"
+
+#include <fstream>
+#include <ftw.h>
+#include <fnmatch.h>
+#include <linux/limits.h>
+#include <unistd.h>
+#include <regex>
+#include <signal.h>
+
+Test::Test* g_plugin;
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+
+NWNX_PLUGIN_ENTRY Plugin::Info* PluginInfo()
+{
+    return new Plugin::Info
+        {
+            "Test",
+            "Test some stuff",
+            "jmd",
+            "jmd2028@gmail.com",
+            1,
+            true,
+            0
+        };
+}
+
+NWNX_PLUGIN_ENTRY Plugin* PluginLoad(Plugin::CreateParams params)
+{
+    g_plugin = new Test::Test(params);
+    return g_plugin;
+}
+
+namespace Test
+{
+
+Test::Test(const Plugin::CreateParams& params)
+        : Plugin(params),
+            m_compiler{std::make_unique<CScriptCompiler>()}
+{
+    m_sourceDir = GetServices()->m_config->Get<std::string>("SOURCE_DIR", "");
+
+    if (m_sourceDir.empty())
+    {
+        // Check if we're symlinked in from build/source tree.  All this would be better with std::filesystem
```

Comment:
A good way to do this is to try and open `__FILE__`.

Reply:
That's a million times better idea,  thank you.

Reply:
I opted to go with just adding a define in cmake.  When i tried `__FILE__` it was giving me relative pathnames into the source tree, I wouldn't have guessed that it would but.. at least the actually code here is a lot clearer and more robust.